app/src/main/java/com/zrcoding/hackertab/core/CustomComponent.kt

```diff
@@ -15,101 +20,109 @@ import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
-import androidx.compose.ui.unit.dp
 import com.zrcoding.hackertab.R
-import com.zrcoding.hackertab.domain.Languages
+import com.zrcoding.hackertab.ui.theme.HackertabTheme
 import com.zrcoding.hackertab.ui.theme.TextLink
+import com.zrcoding.hackertab.ui.theme.dimenBig
+import com.zrcoding.hackertab.ui.theme.dimenLarge
+import com.zrcoding.hackertab.ui.theme.dimenMedium
+
+data class ChipData(
+    val id: String,
+    val name: String,
+    @DrawableRes val image: Int? = null,
+    val selected: Boolean = false
+)
 
-@Preview
 @Composable
 fun Chip(
-    chipModel: ChipData = ChipData("Hello", R.drawable.ic_github),
+    chipData: ChipData,
     isSelected: Boolean = false,
-    onSelectionChanged: (ChipData) -> Unit = {},
+    onClick: (ChipData) -> Unit,
 ) {
     Surface(
-        modifier = Modifier
-            .padding(4.dp)
-            .wrapContentSize(),
-        elevation = 8.dp,
-        shape = RoundedCornerShape(20.dp),
+        shape = RoundedCornerShape(dimenBig),
         color = if (isSelected) TextLink else MaterialTheme.colors.secondary
     ) {
         Row(
             modifier = Modifier
-                .toggleable(
-                    value = isSelected,
-                    onValueChange = {
-                        onSelectionChanged(chipModel)
-                    }
-                ),
+                .padding(horizontal = dimenLarge)
+                .clickable { onClick(chipData) },
             verticalAlignment = Alignment.CenterVertically
         ) {
-            if (isSelected) {
-                Text(
+            chipData.image?.let {
+                Icon(
                     modifier = Modifier
-                        .padding(start = 8.dp)
-                        .height(16.dp)
-                        .width(16.dp),
-                    text = "x"
-                )
-            }
-
-            chipModel.image?.let {
-                Image(
-                    modifier = Modifier
-                        .padding(start = 8.dp)
-                        .height(16.dp)
-                        .width(16.dp),
+                        .padding(end = dimenMedium)
+                        .size(dimenBig),
                     painter = painterResource(id = it),
-                    contentDescription = null
+                    contentDescription = null,
+                    tint = MaterialTheme.colors.surface
                 )
             }
 
             Text(
-                modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp),
-                text = chipModel.name,
+                modifier = Modifier.padding(vertical = dimenMedium),
```

Comment:
@zouhir96 what do you think of using a data class for those dimensions.
I would suggest grouping all the preferred dimensions used in the app in a data class like so:
```kotlin
data class Dimensions(
    val macro: Dp = 2.dp,
    val tiny: Dp = 4.dp,
    val small: Dp = 8.dp,
    val medium: Dp = 16.dp,
    val large: Dp = 24.dp,
    val xlarge: Dp = 32.dp,
    val huge: Dp = 64.dp,
)
``` 
And then you can access it via MaterialTheme using an extension:
```kotlin
val MaterialTheme.dimensions
``` 

As a result, you will have a centralized dimension values.
eg. `MaterialTheme.dimensions.medium`

Reply:
Greaat idea
But i think we can use object instead of data class ?

Reply:
I prefer a data class because the declared dimensions are a type of data. But an object would fit perfectly to 👍🏼